File: app/src/main/java/mvvmkotlin/acaziasoft/app/com/mvvmkotlinexample/manager/FontManager.java
package mvvmkotlin.acaziasoft.app.com.mvvmkotlinexample.manager;

import android.content.Context;
import android.graphics.Typeface;
import android.text.TextUtils;

import java.util.HashMap;

public enum FontManager {
	INSTANCE;

	public static final String DEFAULT_FONT ="opensansregular.ttf";
	private Context mContext;
	private HashMap<String, Typeface> mapTypefaces = new HashMap<>(16);

	public static void init(Context mContext) {
		INSTANCE.mContext = mContext;
	}

	public Typeface setTypeface(String fontName) {
		if (TextUtils.isEmpty(fontName)){
			fontName = DEFAULT_FONT;
		}
		Typeface tf = mapTypefaces.get(fontName);
		if (tf != null) {
			return tf;
		}
		tf = getTypefaceAsset(fontName);
		mapTypefaces.put(fontName, tf);
		return tf;
	}


	public Typeface getTypefaceAsset(String fontName) {
		try {
			return Typeface.createFromAsset(mContext.getAssets(), "fonts/" + fontName);
		} catch (Exception e) {
			return Typeface.DEFAULT;
		}
	}
}

File: app/src/main/java/mvvmkotlin/acaziasoft/app/com/mvvmkotlinexample/manager/FontManager.kt
package mvvmkotlin.acaziasoft.app.com.mvvmkotlinexample.manager

import android.content.Context
import android.graphics.Typeface
import android.text.TextUtils

import java.util.HashMap

enum class FontManager {
    INSTANCE;

    private var mContext: Context? = null
    private val mapTypefaces = HashMap<String, Typeface>(16)

    fun setTypeface(fontName: String): Typeface? {
        var fontName = fontName
        if (TextUtils.isEmpty(fontName)) {
            fontName = DEFAULT_FONT
        }
        var tf: Typeface? = mapTypefaces[fontName]
        if (tf != null) {
            return tf
        }
        tf = getTypefaceAsset(fontName)
        mapTypefaces.put(fontName, tf)
        return tf
    }


    fun getTypefaceAsset(fontName: String): Typeface {
        try {
            return Typeface.createFromAsset(mContext!!.assets, "fonts/" + fontName)
        } catch (e: Exception) {
            return Typeface.DEFAULT
        }

    }

    companion object {

        val DEFAULT_FONT = "opensansregular.ttf"

        fun init(mContext: Context) {
            INSTANCE.mContext = mContext
        }
    }
}